packages/webview_flutter/lib/src/tizen_webview.dart

```diff
@@ -0,0 +1,228 @@
+
+/// Signature for callbacks that report a method calls by navigation delegate method channel.
+typedef NavigationDelegateMethodChannelCallback = Future<dynamic> Function(
+    MethodCall method);
+
+/// A Tizen webview that displays web pages.
+class TizenWebview {
+  /// Constructs a new WebView.
+  TizenWebview()
+      : _controllerChannel =
+            const MethodChannel('plugins.flutter.io/tizen_webview_controller'),
+        _navigationDelegateChannel = const MethodChannel(
+            'plugins.flutter.io/tizen_webview_navigation_delegate') {
+    _controllerChannel.setMethodCallHandler(_onMethodCall);
+  }
+
+  final MethodChannel _controllerChannel;
+  final MethodChannel _navigationDelegateChannel;
+  bool _isCreated = false;
+  bool _hasNavigationDelegate = false;
+  final Map<String, JavaScriptChannelParams> _javaScriptChannelParams =
+      <String, JavaScriptChannelParams>{};
+  final Map<String, dynamic> _pendingMethodCalls = <String, dynamic>{};
+
+  Future<bool?> _onMethodCall(MethodCall call) async {
+    switch (call.method) {
+      case 'javaScriptChannelMessage':
+        final String channel = call.arguments['channel']! as String;
+        final String message = call.arguments['message']! as String;
+        if (_javaScriptChannelParams.containsKey(channel)) {
+          _javaScriptChannelParams[channel]
+              ?.onMessageReceived(JavaScriptMessage(message: message));
+        }
+
+        return true;
+    }
+
+    throw MissingPluginException(
+      '${call.method} was invoked but has no handler',
+    );
+  }
+
+  Future<T?> _invokeChannelMethod<T>(String method, [dynamic arguments]) async {
+    if (!_isCreated) {
+      _pendingMethodCalls[method] = arguments;
+      return Future<T?>(() => null);
+    }
+
+    return _controllerChannel.invokeMethod<T>(method, arguments);
+  }
+
+  /// Called when [TizenView] is created. Invokes the requested method call before [TizenWebView] is created.
+  void onCreate() {
+    _isCreated = true;
+    _callPendingMethodCalls();
+  }
+
+  /// Set the method call handler of the navigation delegate channel.
+  void setNavigationDelegateMethodCallHandler(
+      NavigationDelegateMethodChannelCallback method) {
+    _navigationDelegateChannel.setMethodCallHandler(method);
+  }
+
+  /// Applies the requested settings before [TizenView] is created.
+  void _callPendingMethodCalls() {
+    if (_hasNavigationDelegate) {
+      _invokeChannelMethod<void>(
+          'setNavigationDelegate', _hasNavigationDelegate);
+    }
+
+    _pendingMethodCalls.forEach((String method, dynamic arguments) {
+      _controllerChannel.invokeMethod<void>(method, arguments);
+    });
+    _pendingMethodCalls.clear();
+  }
+
+  /// Loads the file located on the specified [absoluteFilePath].
+  Future<void> loadFile(String absoluteFilePath) async {
+    assert(absoluteFilePath != null);
+    _invokeChannelMethod<void>('loadFile', absoluteFilePath);
```

Comment:
Async methods should be awaited or the returned Future should be returned by the caller. i.e.

```suggestion
    await _invokeChannelMethod<void>('loadFile', absoluteFilePath);
```
or
```suggestion
   return _invokeChannelMethod<void>('loadFile', absoluteFilePath);
```

In the latter case the `async` keyword for the caller method is optional.

Reply:
I have modified all related codes.